### ISSUE TITLE
Alguém sabe / já fez algo parecido?

### ISSUE THREAD

**Issue by vmnog** [vmnog/phpeiros-loja#11]
### Gostaria de começar a fazer mudanças severas no projeto.

1. A tela inicial vai ser uma tela de login
2. Teremos tres tipos de usuários (Owner, Employee, User)
3. Dependendo do tipo de usuário ele mostrará uma tela diferente, com funcionalidades diferentes

Estava pensando em fazer isso e acho que eu faria assim:

- Criaria uma pasta chamada :file_folder:  views /
- Dentro dessa :open_file_folder:  teríamos +3 :file_folder:  pastas

 l___:file_folder:  views /
l________1. :file_folder: owner
l________2.  :file_folder: employee
l________3.  :file_folder: user (ou client, ainda nao decidi o melhor nome)

Dai no login do usuário seria feito essa verificação do tipo da conta,
dependendo do tipo da conta o usuário é redirecionado pra pasta determinada.

> Teria que ser feito uma verificação do tipo de conta também em todas as páginas para que um usuário comúm não consiga, por exemplo, acessar uma página de admin (caso ele tenha o link).

Enfim, é uma ideia, alguém sabe se tem outra forma de fazer isso ou já trabalhou nisso?

**Comment by rodrigosousadf** [comment]
no meu de estudo tava fazendo tipo isso. a tela inicial é um login de acesso restrito que iria indicar qual nível de acesso do usuário e assim liberar as funcionalidades.

**Comment by rodrigosousadf** [comment]
já meio que saquei aqui, tu tem que gerar um nível no usuario, fazer outra coluna no banco de dados.
ai tem que  usar session pra uso das páginas, indicando qual nível dele. ai com isso acho que rola de restringir o acesso a algumas funcionalidades

**Comment by vmnog** [comment]
Exatamente, o que precisamos para começar a implementar?

**Comment by maicomferre** [comment]
Então, no sistema que eu coloquei, ele tem o campo "cargo" no banco de dados, que define se ele é usuário comum ou não, ele começa com um que mentalizei como "admin", mas deve começar com 0 quando a conta é criada para definir como usuário comum. Esta informação pode ser obtida quando ele faz login para fazer as restrições(caso não seja adm).


O "cargo", ficando assim:
Owner = 2
Employee = 1
User = 0

**Comment by maicomferre** [comment]
> * Criaria uma pasta chamada  views
> * Dentro dessa pasta  teríamos +3  pastas
> 
> l___  views
> l________1.  owner
> l________2.   employee
> l________3.   user (ou client, ainda nao decidi o melhor nome)


### Fico pensando aqui. ###
Para mudar isso seria necessário remover a cabeçalho e o rodapé como includes e colocar individualmente nos arquivos. Porque isso causa o problema com o link relativo.

ja que o **cabecalho.php** inicialmente estava apontando para o diretório "raiz" do projeto, mas não do servidor considerando que o projeto fica em /phpeiros-loja/...
e quando eu coloquei subpastas eu criei um outro **cabecalho.2.php** para corrigir o erro. Ele apenas voltava uma pasta anterior para poder acessar os arquivos.

Dai teria o erro, já que teria que criar um cabecalho.3.php e se tiver outra subpasta tendo que criar outro e outro. 

Outro problema que se cria com o uso do cabecalho, ao meu ver, é que se perde a independência do <head> em cada arquivo, ficando necessário que uma alteração em um arquivo, no head, seja feita sempre globalmente.


_Demonstração do uso do cabeçalho:_
A = - > **Pasta que pega o cabecalho.php e rodape.php**
B = - > **Pasta que pega o cabecalho.2.php e rodape.2.php**
M:\XAMPP\HTDOCS\PHPEIROS-LOJA
│   .gitignore
│   categoria-lista.php - > _A_
│   dev.ignore
│   index.php - > _A_
│   README.md
│   visualiza-produto.php - > _A_
│
├───.idea
│       loja.iml
│       misc.xml
│       modules.xml
│       vcs.xml
│       workspace.xml
│
├───admin - > _B_
│       adiciona-categoria.php
│       adiciona-produto.php
│       altera-categoria.php
│       altera-produto.php
│       categoria-altera-formulario.php
│       categoria-formulario.php
│       categoria-lista.php
│       produto-altera-formulario.php
│       produto-formulario.php
│       produto-imagem-reparar.php
│       produto-lista.php
│       remove-categoria.php
│       remove-produto.php
│       reparar-imagem.php
│       visualiza-produto.php
│
├───cadastro - > _B_
│       index.php
│
├───github
│   └───img
│           social-preview.PNG
│
├───login - > _B_
│       index.php
│
├───minhaconta - > _B_
│       index.php
│
└───src
    │   processa-imagem.php
    │
    ├───bancos
    │       banco-categoria.php
    │       banco-produto.php
    │       conecta.php
    │       global.php
    │       pdo-categoria.php
    │       pdo-conecta.php
    │       pdo-produto.php
    │
    ├───css
    │       bootstrap-grid.css
    │       bootstrap-grid.css.map
    │       bootstrap-grid.min.css
    │       bootstrap-grid.min.css.map
    │       bootstrap-reboot.css
    │       bootstrap-reboot.css.map
    │       bootstrap-reboot.min.css
    │       bootstrap-reboot.min.css.map
    │       bootstrap.css
    │       bootstrap.css.map
    │       bootstrap.min.css
    │       bootstrap.min.css.map
    │       loja.css
    │
    ├───includes
    │       cabecalho.2.php
    │       cabecalho.php
    │       rodape.2.php
    │       rodape.php
    │
    ├───js
    │       bootstrap.bundle.js
    |        _+ 8 Arquivos .js_
    ├───produtos
    │   └───img
    │           default.png
    │
    ├───servidor
    │       cadastrar.php
    │       funcoes-sessao.php
    │       logado.php
    │       logar.php
    │       sair.php
    │       usuario.php
    │
/////Continua...

Creio que apenas 3  elementos alterados entre cabeçalho.php e cabeçalho.2.php.
Exemplo:
Mudando só de:
><script src="src/js/produto.js"></script>
Para
> <script src="../src/js/produto.js"></script>

Bom, não tenho certeza se há algo melhor, minha sugestão é mudar tudo para "cada script com seu code ao invés de include", mas pode haver forma melhor.

**Comment by vmnog** [comment]
Acho que seria melhor recomeçarmos, o projeto, mas se preocupando mais com a funcionalidade e menos com aparencia.

Definir estrutura mvc,
nomes dos arquivos,
onde vai ficar cada arquivo,
funcionalidades basicas e essenciais
banco de dados.

O que acham?

**Comment by maicomferre** [comment]
Exato, seria um trabalho grande, mas não diferencia muito da grandeza que é organizar os arquivos atuais.

**Comment by vmnog** [comment]
eu acho que tentar adaptar o que temos agora da mais trabalho por causa do front end, uma vez que conseguirmos começar do jeito certo ficaria mais facil fazer alterações e adaptações
